src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,1002 @@ const ActorInit Dm_Char08_InitVars = {
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFA18.s")
+void func_80AAFB94(DmChar08* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAC4.s")
+Color_RGBA8 sBubblePrimColor = { 0xFA, 0xC8, 0xFA, 0x64 };
+Color_RGBA8 sBubbleEnvColor = { 0x50, 0x50, 0x50, 0 };
+Vec3f sBubbelAccel = { 0.0f, -0.8f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAE4.s")
+void func_80AAFBA4(DmChar08* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f velocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFB04.s")
+    velocity.x = (Rand_ZeroOne() - 0.5f) * 24.0f;
+    velocity.y = -5.2f;
+    velocity.z = 4.0f;
+    sBubbelAccel.x = 0.0f;
+    sBubbelAccel.y = 1.18f;
+    sBubbelAccel.z = 0.0f;
+    for (i = 0; i < this->bubbleCount; i++) {
+        EffectSsDtBubble_SpawnCustomColor(globalCtx, &this->bubblePos, &velocity, &sBubbelAccel, &sBubblePrimColor,
+                                          &sBubbleEnvColor, Rand_S16Offset(80, 130), 30, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFB94.s")
+void func_80AAFCCC(DmChar08* this, GlobalContext* globalCtx) {
+    switch (globalCtx->sceneNum) {
+        case SCENE_31MISAKI:
+            if (!(gSaveContext.save.weekEventReg[0x37] & 0x80)) {
+                switch (this->unk_206) {
+                    case 0:
+                        return; // Must be a return
+                    case 1:
+                        Message_StartTextbox(globalCtx, 0x102A, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+                    case 2:
+                        if (Message_GetState(&globalCtx->msgCtx) == 2) {
+                            Message_StartTextbox(globalCtx, 0x102B, &this->dyna.actor);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            } else {
+                switch (this->unk_206) {
+                    case 0:
+                        break;
+                    case 1:
+                        Message_StartTextbox(globalCtx, 0x103B, &this->dyna.actor);
+                        this->unk_206++;
+                        break;
+                    case 2:
+                        if (Message_GetState(&globalCtx->msgCtx) == 2) {
+                            func_801477B4(globalCtx);
+                            this->unk_206 = 0;
+                        }
+                        break;
+                }
+            }
+            break;
+        case SCENE_SEA:
+            switch (this->unk_206) {
+                case 0:
+                    break;
+                case 1:
+                    Message_StartTextbox(globalCtx, 0x102E, &this->dyna.actor);
+                    this->unk_206++;
+                    break;
+                    ;
```

Comment:
Necessary? 
Seems like there are more of these down below. Are they all necessary?